packages/arb-validator-core/message/transactionBatch.go

```diff
@@ -0,0 +1,315 @@
+/*
+* Copyright 2020, Offchain Labs, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package message
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-util/common"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/hashing"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/value"
+)
+
+type TransactionBatch struct {
+	Chain        common.Address
+	Tos          []common.Address
+	SequenceNums []*big.Int
+	Values       []*big.Int
+	DataLengths  []uint32
+	Data         []byte
+	Signatures   []byte
+}
+
+func OffchainTxHash(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [exported function OffchainTxHash should have comment or be unexported](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)